std/internal/math/biguintcore.d

```diff
@@ -1113,6 +1113,27 @@ public:
         return .hashOf(data);
     }
 
+    /**
+    Random generate of the number of specified uintLength.
+    Params:
+        uintLength = the specified uintLength.
+    */
+    void randomGenerate(const size_t uintLength) @safe
+    {
+        assert(uintLength > 0);
+
+        import std.random : rndGen, uniform;
+        auto rnd = rndGen;
```

Comment:
You don't want to assign rndGen to a local variable, because that actually creates a **copy** of `rndGen`. That would cause this function to repeatedly return the same result.

The design of `rndGen` and `std.random` in general make it easy to accidentally create a copy. That's why @9il's mir-random generators disallow implicit copies via `@disable this(this)`, which initially seemed draconian to me but I am coming to agree with.

Reply:
Sorry, now I understand.